### PR TITLE
server/onesync: three new 'state' natives

### PR DESCRIPTION
**Pull request by PichotM**
Useful for some server/entities stuff.
`HAS_VEHICLE_BEEN_OWNED_BY_PLAYER`
`HAS_ENTITY_BEEN_MARKED_AS_NO_LONGER_NEEDED`
This one only works with vehicle since ped/object game state nodes still need to be made.

### CODE REVIEW

**Comment by smallo92** [comment]
Do you mean HAS_ENTITY_BEEN_MARKED_AS_NO_LONGER_NEEDED not HAS_ENTITY_BEEN_MARKED_HAS_NO_LONGER_NEEDED

**Comment by PichotM** [comment]
Indeed, I don't know why I did that
Thanks

**Comment by blattersturm** [comment]
Did this get a natives PR?

File: code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp [hunk 1]
```diff
@@ -463,4 +432,45 @@ static InitFunction initFunction([]()
...
+
+		// get the owning server instance
+		auto instance = resourceManager->GetComponent<fx::ServerInstanceBaseRef>()->Get();
+
+		// get the server's game state
+		auto gameState = instance->GetComponent<fx::ServerGameState>();
+
+		std::vector<int> entityList;
+
+		for (auto& entity : gameState->m_entityList)
```
Comment (blattersturm) [inline code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp]: Should lock m_entityListMutex.

Comment (PichotM) [reply code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp]: Should be fine now

**Comment by PichotM** [comment]
> Did this get a natives PR?

Yes. https://github.com/citizenfx/natives/commit/0064abe827364b106674d355e1e523b59c4355a8

**Comment by PichotM** [comment]
hello

**Comment by blattersturm** [comment]
meh another rebase needed :(

**Comment by PichotM** [comment]
Commit history is super ugly, but conflicts have been fixed

**Comment by summeryukata** [comment]
https://stackoverflow.com/a/5201642

**Comment by PichotM** [comment]
if it breaks the build, I'll write 2 DOCS

**Pull request merged**